### DISCUSSION TITLE
StrongSwan's performance deteriorated?

### DISCUSSION THREAD

**Discussion by onoder** [strongswan/strongswan#1970] [General]
In my project, I am using StrongSwan for the IKEv1 implementation.
Please teach me about the following:
[Background]
In my project, the initiator is requesting the responder to send between 2000 and 4000 routes.
Our test results showed that StrongSwan's performance deteriorated when transmitting on 2000 and 4000 routes compared to when it was performed on 1000 routes.
[Question]
1) Is this kind of use of StrongSwan unexpected?
2) If you know change the settings to improve the performance of StrongSwan, please teach me.

**Comment by tobiasbrunner** [comment]
What exactly do you mean with "routes"? And when and how does the performance deteriorate?

**Comment by onoder** [comment]
Thank you for your response.
I use a translation app. The translation app did not convert as expected.
"routes" refers to VPN tunnels
We measured the time it took to connect with "1000 VPN tunnels" and "2000 VPN tunnels".
"2000VPN tunnel" took more than twice as long as "1000VPN tunnel".

**Reply by tobiasbrunner** [reply]
Still doesn't fully clarify what you mean, in particular considering the difference between IKE and Child SAs, or Main and Quick Mode in IKEv1. So what exactly did you configure, what exactly is negotiated, and, again, when and how exactly does the performance deteriorate.

And please don't use IKEv1 anymore.
